app/src/main/java/org/stepik/android/presentation/course_search/CourseSearchFeature.kt

```diff
@@ -0,0 +1,41 @@
+package org.stepik.android.presentation.course_search
+
+import org.stepik.android.domain.course_search.model.CourseSearchResultListItem
+import org.stepik.android.model.Step
+import org.stepik.android.model.comments.DiscussionThread
+import ru.nobird.android.core.model.PagedList
+
+interface CourseSearchFeature {
+    sealed class State {
+        object Idle : State()
+        object Loading : State()
+        object Empty : State()
+        object Error : State()
+        data class Content(
+            val courseSearchResultListDataItems: PagedList<CourseSearchResultListItem.Data>,
+            val courseSearchResultListItems: List<CourseSearchResultListItem>
```

Comment:
lets determine loading next page state with `Boolean` field instead of list copy